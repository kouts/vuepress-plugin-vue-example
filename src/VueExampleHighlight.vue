<template>
  <div ref="formattedCode" :class="['shiki-container', `shiki-${language}`]"></div>
</template>

<script>
import { codeToHtml } from 'shiki'

export default {
  name: 'VueExampleHighlight',
  props: {
    code: {
      type: String,
      default: '',
    },
    language: {
      type: String,
      default: 'vue-html',
    },
  },
  mounted: async function () {
    const res = await codeToHtml(this.$props.code, {
      lang: this.$props.language,
      theme: 'github-dark-default',
    })

    const codeEl = document.createElement('div')

    codeEl.innerHTML = res

    this.$refs.formattedCode.appendChild(codeEl)
  },
}
</script>

<style lang="scss" scoped>
.shiki-container :deep(pre) {
  padding: 0;
  margin: 0;
  overflow: auto;
  line-height: 1.4;
  padding: 1.25rem 1.5rem;
  border-radius: 6px;
  font-size: 14px;
}

.shiki-container :deep(pre > code) {
  background-color: transparent;
}
</style>
