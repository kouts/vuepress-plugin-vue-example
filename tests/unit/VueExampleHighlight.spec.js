import { shallowMount } from '@vue/test-utils';
import VueExampleHighlight from '@/VueExampleHighlight';
import Prism from 'prismjs';
jest.mock('prismjs');
Prism.highlight = jest.fn((code) => code);

describe('VueExampleHighlight', () => {

  const props = {
    code: '<p>Hello world</p>',
    language: 'html'
  };

  it('renders the correct markup', () => {
    const wrapper = shallowMount(VueExampleHighlight, {
      propsData: props
    });
    const div = wrapper.find(`div.language-${wrapper.props().language}`);
    const pre = wrapper.find(`pre.language-${wrapper.props().language}`);
    const code = wrapper.find('code');
    expect(div.exists()).toBe(true);
    expect(pre.exists()).toBe(true);
    expect(code.exists()).toBe(true);
    expect(code.text()).toContain('Hello world');
    // console.log(wrapper.html());
  });

  it('renders the correctly', () => {
    const wrapper = shallowMount(VueExampleHighlight, {
      propsData: props
    });
    expect(wrapper).toMatchSnapshot();
  });  

});
