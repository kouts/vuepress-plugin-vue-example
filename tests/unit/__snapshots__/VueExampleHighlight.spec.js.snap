// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueExampleHighlight > renders the correctly 1`] = `
"<div data-v-e6ee2242="" class="language-html"><pre data-v-e6ee2242="" class="language-html"><code><p>Hello world</p></code></pre>
</div>"
`;
